--- jupyter-compute	1970-01-01 12:00:00.000000000 +1200
+++ jupyter-compute	2019-08-15 14:35:38.087759000 +1200
@@ -0,0 +1,14 @@
+#!/bin/env bash
+# NeSI 2019
+# this tool provides portforwarding from the submitted node to the compute node and
+# starts jupyter lab
+#  mandes.schoenherr@nesi.org.nz
+
+port=${1:-15051}
+
+echo "Jupyter lab will be started on $(hostname) and establish port farwarding from $SLURM_SUBMIT_HOST on port $port"
+
+ssh -N -R ${port}:localhost:${port} $SLURM_SUBMIT_HOST &
+
+XDG_RUNTIME_DIR="" jupyter lab --NotebookApp.ip='0.0.0.0' --port ${port} --no-browser ${@:2}
+
